### PR TITLE
Codebase rework

### PR DESCRIPTION
**Pull request by splovyt**
I have worked on the following:

- Dockerfile: use max-base as base-image to speed up the image and remove some of the requirements
- README: correct a typo
- refactored the `app.py` and `chatbot.py` code
- added/moved the license headers where necessary
- added documentation to the code to help understand what's going on
- made some minor tweaks to make the code more efficient
- updated to Bootstrap 4
- made the frontend look a lot better by playing with whitespace, adding descriptions, links, and move to the IBM Plex font

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -15,48 +15,59 @@
 <html>
 <head>
     <title>MAX Question Answering Chatbot Demo</title>
-    <link rel="stylesheet" href="static/css/bootstrap.min.css">
+    <link rel="stylesheet" href="static/css/bootstrap.css">
```
Comment (bdwyer2) [inline templates/index.html:18]: What is the reason for this change?

Comment (splovyt) [reply templates/index.html:18]: I updated to the latest version of Boostrap (4), and I prefer the non-minimized version cause it's easier to debug and alter for other people.. Given the scale that we work at the loading perfomance is negligible

**Review by kmh4321 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Create a machine learning powered web app to answer questions
 
-Ever found yourself wondering what the mitochondria is? Perhaps you are curious about how neurons communicate with each other? A Google search works well to answer your questions, but how about something still digestible, but more precise? This code pattern will show you how to build a chatbot that will answer questions found in a college biology textbook. In this pattern, the textbook used is [Biology 2e](https://books.google.com/books/about/Biology_2e.html?id=JLPQtwEACAAJ) by Mary Ann Clark, Matthew Douglas, and Jung Choi. 
+Ever found yourself wondering what the mitochondria are? Perhaps you are curious about how neurons communicate with each other? A Google search works well to answer your questions, but how about something still digestible, but more precise? This code pattern will show you how to build a chatbot that will answer questions found in a college biology textbook. In this pattern, the textbook used is [Biology 2e](https://books.google.com/books/about/Biology_2e.html?id=JLPQtwEACAAJ) by Mary Ann Clark, Matthew Douglas, and Jung Choi.
```
Comment (djalova) [inline README.md]: "what mitochondria are" sounds better to me

Comment (splovyt) [reply README.md]: Yeah you are right, thanks1

File: chatbot.py [hunk 3]
```diff
@@ -38,28 +39,26 @@ def get_close_matches(topic, titles):
...
+    if topic.lower() == "stop":
+        return "Thank You for using QnAit, Hope your questions were answered!", 5, {}
+
     # hardcoded fun :)
     if topic == "What is the meaning of life?":
-        return "42\nIs there another topic you are curious about?", 1, {}
-    
-    if topic == "No":
-        return "Thank You for using QnAit, Hope your questions were answered!", 5, {}
+        return "42\n\nIf you are curious about another topic, reply with the topic.", 1, {
```
Comment (djalova) [inline chatbot.py]: I think this `{` doesn't have a `}` to close it

File: templates/index.html [hunk 4]
```diff
@@ -15,48 +15,59 @@
...
+    </section>
 </body>
 
 <script type="text/javascript">
     let state = 1
     let matches = []
-    const defaultLoadingMessage = "QnAIT is thinking"
-    const secretLoadingMessages = ['Hold on, QnAIT is skimming the textbook for you', 'Give us a second, our bot escaped', 'Our intern is typing your answer for you']
+    const defaultLoadingMessage = "QnAIT is thinking..."
+    const secretLoadingMessages = ['Hold on, QnAIT is skimming the textbook for you...', 'Give us a second, our bot escaped...', 'Our intern is typing your answer for you...']
```
Comment (djalova) [inline templates/index.html]: You don't need to add ellipsis here...

File: chatbot.py (hunk 3 above)
Comment (splovyt) [reply chatbot.py]: Thanks I verified it runs now

**Review by kmh4321 — DISMISSED** [review]

**Review by djalova — DISMISSED** [review]

**Review by kmh4321 — APPROVED** [review]
